common/src/messaging/mod.rs

```diff
@@ -0,0 +1,15 @@
+use async_trait::async_trait;
+
+mod crossbeam_messaging;
+
+pub use crossbeam_messaging::{CrossbeamReceiver, CrossbeamSender};
+
+#[async_trait]
+pub trait AsyncReceiver<T, E: std::error::Error> {
```

Comment:
std Error => could be our own error or any anyhow one

Reply:
We leave the door open without restraining too much on functionalities using std::error::Error imo